cpp/src/gandiva/engine.cc

```diff
@@ -303,12 +302,18 @@ Status Engine::FinalizeModule() {
 
   ARROW_RETURN_IF(llvm::verifyModule(*module_, &llvm::errs()),
                   Status::CodeGenError("Module verification failed after optimizer"));
-
   // do the compilation
-  execution_engine_->finalizeObject();
-  module_finalized_ = true;
+  if(execution_engine_->hasError()) {
+    //execution_engine_->finalizeObject();
+    ARROW_LOG(INFO) << "[OBJ-CACHE-LOG][ERROR]: " << execution_engine_->getErrorMessage();
+    module_finalized_ = false;
+    return Status::OK();
+  } else {
+    execution_engine_->finalizeObject();
+    module_finalized_ = true;
+    return Status::OK();
+  }
```

Comment:
I think you can move the code outside the `else` block and remove the `else`

Reply:
Done